site-src/geps/gep-1619.md

```diff
@@ -0,0 +1,84 @@
+# GEP-1619: Session Persistence and Session Affinity
+
+* Issue: [#1619](https://github.com/kubernetes-sigs/gateway-api/issues/1619)
+* Status: Provisional
+
+(See definitions in [GEP Status][/contributing/gep#status].)
+
+## TLDR
+
+This GEP proposes a definition of session persistence and session affinity as the foundation of future session persistence related improvements. This GEP also outlines the ways session persistence could be achieved by implementations.
+
+## Goals
+- Define session persistence and session affinity to establish a common language
+- Identify differences in session persistence functionality between implementations
+
+## Non-Goals
+- Create proposals for adding session persistence configuration to Gateway API object specification
+- Mandate a default or expected session persistence functionality for implementations
+- Require implementations to support session persistence or a specific form of it
+
+## Introduction
+
+At the November 28th, 2022 SIG Gateway API meeting, we discussed the idea of standardizing and documenting session persistence (also known as session stickiness). It was recommended that we first generate discussion for how we describe session persistence, agree on how it is achieved, and find out which implementers support session persistence. 
+
+Conformance test for session persistence will be considered if deemed appropriate. Furthermore, this information can be used for developing a Gateway API spec to configure session persistence in the future.
+
+### Defining Session Persistence
+
+Session persistence is the act of using application layer information to direct a client request to the same backend server for the duration of a "session". Session persistence, when properly configured, guarantees a user will be directed to the same server. Persistence is an exception to load balancing: a persistent client request bypasses the proxy's load balancing algorithm, going directly to a backend server it has previously established a session with.
+
+Some of the concerns of session persistence are the duration and expiration of the session, security of the transaction stream, and storage of the context or state. Application developers use session persistence to improve the performance and reduce overall storage needs by aligning each transaction and its cached data with the same server.
+
+Session persistence is not to be confused with session affinity, which uses information below the application layer to maintain a client request to a single server. Session affinity can be considered a weaker form of session persistence: it is not guaranteed to persist a connection to the same backend server.
```

Comment:
This seems to be saying the difference between Affinity and Persistence is L7 vs L4. Its not clear to me that is an important distinction

In my view, the primary difference between session types is whether the client _directly_ specifies the backend they want OR if some attribute of the request is just used to consistently send to a backend.

* Type 1 example: A cookie with `Backend=1.2.3.4`
* Type 2 example: Send to backend based on `backend[hash(SomeHeader) % BackendCount]`

In Envoy terms, 1 is "stick sessions" and 2 is "consistent hash".

To me, that distinction seems more relevant than L7 vs L4. I would agree that "affinity" does tend to be L4 based on source IP, as you noted in your examples.

Reply:
Good point, I agree. I reworded Persistence and Affinity verbiage with focus on what information/attributes are being used to achieve the connection consistency, let me know what you think.